### ISSUE TITLE
Creating an optimized production build strips out `wheel` event listener from 13.4.10-canary.1 onwards

### ISSUE THREAD

**Issue by i-am-the-slime** [vercel/next.js#53429]
### Verify canary release

- [X] I verified that the issue exists in the latest Next.js canary release

### Provide environment information

```bash
Operating System:
      Platform: darwin
      Arch: arm64
      Version: Darwin Kernel Version 22.5.0: Thu Jun  8 22:22:20 PDT 2023; root:xnu-8796.121.3~7/RELEASE_ARM64_T6000
    Binaries:
      Node: 16.19.1
      npm: 8.19.3
      Yarn: 1.22.10
      pnpm: 8.6.9
    Relevant Packages:
      next: 13.4.13-canary.8
      eslint-config-next: 13.4.12
      react: 18.2.0
      react-dom: 18.2.0
      typescript: 5.1.6
    Next.js Config:
      output: N/A
```

### Link to the code that reproduces this issue or a replay of the bug

You'll have to trust me

### To Reproduce

Use the library `tldraw/core` which in turn depends on `useGesture/react` and provide an onZoom handler to the `Renderer`.
Run `next dev` and see that everything works.
Run `next build && next start` and zooming and panning don't work anymore.

### Describe the Bug

In dev and versions up until and including: `13.4.10-canary.0` there will be a `wheel` eventlistener on the canvas. 
But when doing `next build` in the final version this handler is somehow optimised away.

Therefore, the bug must have been introduced in 13.4.10-canary.1:

https://github.com/vercel/next.js/releases/tag/v13.4.10-canary.1

### Expected Behavior

Eventlisteners don't get deleted.

**Comment by icyJoseph** [comment]
Hi,

There had been some SWC related issues with code getting dropped during minification.

If you disable swcMinify, by setting it to false in `next.config.js`, does it work?

**Comment by i-am-the-slime** [comment]
Yes, that helps.
To clarify, now it's still minified but more slowly and correctly with terser, right?

**Comment by icyJoseph** [comment]
Yes, but AFAIK, there's a fix merged to upstream SWC.

- https://github.com/vercel/next.js/pull/53308 -> fix
- https://github.com/vercel/next.js/pull/53381 -> fix reverted

😅 I'd say, check again when next@12.4.13 or whatever follows comes

**Comment by i-am-the-slime** [comment]
Cool! I was hoping this would magically fix itself but I felt I should at least report something 🍔 !

**Comment by icyJoseph** [comment]
Yes, I mean, the update to swc was reverted.... so it is probably a good idea that the issue was re-opened.

**Comment by i-am-the-slime** [comment]
Yeah, well no, I will definitely not do more work for free on this. It was annoying enough to binary search the breaking version and do my write-up.
I'll definitely won't report any bugs in the future.
